src/audio/google_rtc_audio_processing.c

```diff
@@ -225,14 +330,36 @@ static struct comp_dev *google_rtc_audio_processing_create(
 	if (!cd->tuning_handler)
 		goto fail;
 
-	cd->state = GoogleRtcAudioProcessingCreate();
+	if (MEMORY_BUFFER_BYTES > 0) {
+		cd->memory_buffer = rballoc(0, SOF_MEM_CAPS_RAM,
+					    MEMORY_BUFFER_BYTES * sizeof(cd->memory_buffer[0]));
+		if (!cd->memory_buffer) {
+			comp_err(dev, "google_rtc_audio_processing_create: failed to allocate memory buffer");
+
+			goto fail;
+		}
+
+		GoogleRtcAudioProcessingAttachMemoryBuffer(cd->memory_buffer, MEMORY_BUFFER_BYTES);
+	} else {
+		cd->memory_buffer = NULL;
```

Comment:
this should be impossible to get to, and if you did modify ti to be accessible you will leak here

Reply:
I don't follow.
If MEMORY_BUFFER is zero, this simply initializes memory_buffer to NULL, and no allocation is done of memory_buffer. Therefore there should not be any leakage.

Reply:
Nice! I removed that.